readthedocs/proxito/middleware.py

```diff
@@ -120,7 +120,7 @@ class ProxitoMiddleware(MiddlewareMixin):
     def add_proxito_headers(self, request, response):
         """Add debugging headers to proxito responses."""
 
-        project_slug = getattr(request, 'host_project_slug', '')
+        project_slug = getattr(request, 'path_project_slug', '')
```

Comment:
Do we not need to use `host_project_slug` sometimes as well? It seems like this is breaking existing logic, unless `path_project_slug` is getting set other places?

Reply:
> Do we not need to use `host_project_slug` sometimes as well? 

As a fallback, you mean? In case that `path_project_slug` is not available? Why it would not be available?

> It seems like this is breaking existing logic, unless `path_project_slug` is getting set other places?

What existing logic do you think it's breaking? If you have an example on mind I can try to write a test case.

Reply:
We are using the same logic for `version_path_project` that follows this line. It's used here and set in `_get_project_data_from_request` (https://github.com/readthedocs/readthedocs.org/blob/a81d82ed736b611491050f39649bd668ef212c0f/readthedocs/proxito/views/utils.py#L96) which is the first function that is called when serving a page at https://github.com/readthedocs/readthedocs.org/blob/a81d82ed736b611491050f39649bd668ef212c0f/readthedocs/proxito/views/serve.py#L72-L79

The `host_project_slug` is returned by `map_host_project_slug` which is accurate, https://github.com/readthedocs/readthedocs.org/blob/a81d82ed736b611491050f39649bd668ef212c0f/readthedocs/proxito/middleware.py#L159. However, the project slug we get from the host _sometimes_ (in subprojects or translations, for example) is not the project slug being served.

Reply:
Gotcha, are other uses of `host_project_slug` unaffected by this issue? It just seems bad to have 2 different variables for the project_slug, where we use them in different places.

Reply:
> It just seems bad to have 2 different variables for the project_slug, where we use them in different places.

Well, if I'm understand correctly this they _are different things_:

* `host_project_slug` is the project slug that we infer from the host/domain's URL
  * not accurate for subdomain/translations
  * it's fine for custom domains and `.readthedocs.io`
  * it's set at the beginning of the request in the middleware
* `path_project_slug` is the project slug that we can infer from the path's URL mainly
  * it's good for subdomain/translations
  * it's the same as what we have been calling `final_project` (maybe it's better to call it `final_project_slug`?)
  * it's set in the `_get_project_data_from_request` which is called from inside the `ServeDocs` view in Proxito as the first step of the request after all the middlewares

It seems that _both_ are required. We use `host_project_slug` as first step trying to get the `Project` object (`map_project_slug` decorator). Then, inside the `_get_project_data_from_request` we use `host_project_slug` to check single project, translation and subproject and set the `path_project_slug` which in those cases could be different than `host_project_slug`.

> Gotcha, are other uses of `host_project_slug` unaffected by this issue? 

Grepping the code I didn't find anything that clearly needs to be adapted. Maybe there is something deeper that I'm not seeing, tho

Reply:
This makes sense to me. We should probably document all these variables (as you have done above) in the code/docs, so we don't get confused in the future.